src/c++/perf_analyzer/docs/examples/profile.py

```diff
@@ -0,0 +1,154 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import json
+import subprocess
+from pathlib import Path
+
+TEMP_INPUT_FILE = "temp_input_data.json"
+
+
+def calculate_avg_latencies():
+    # Example json demonstrating format:
+    #   see client/src/c++/perf_analyzer/docs/examples/decoupled_output_file.json
+    first_token_latencies = []
+    token_to_token_latencies = []
+    with open("profile_export.json") as f:
+        requests = json.load(f)["experiments"][0]["requests"]
+        for request in requests:
+            prev_response = request["response_timestamps"][0]
+            first_token_latencies.append(prev_response - request["timestamp"])
+            for response in request["response_timestamps"][1:]:
+                token_to_token_latencies.append(response - prev_response)
+                prev_response = response
+
+    avg_first_token_latency = (
```

Comment:
Redundant code/formula. Should be refactored into a method. Or, might there be a library that does this for us?

Reply:
Is there a way to convert from nanoseconds to seconds using a library? This would remove the need for a comment. I found this: [https://www.geeksforgeeks.org/python-timedelta-total_seconds-method-with-example/](https://www.geeksforgeeks.org/python-timedelta-total_seconds-method-with-example/)

Reply:
I think the code is pretty straightforward. Given that this code is used only for our guide and nowhere in our codebase, I don't think there's much value on refactoring further.